deltachat-ios/View/TextFieldCell.swift

```diff
@@ -1,37 +1,87 @@
+        }
+        get {
+            return placeholderVal
+        }
+    }
+
+    private var fontSize: CGFloat {
+        return UIFont.preferredFont(forTextStyle: .body).pointSize
+    }
+
+    private var preferValue: Bool {
+        set {
+            textField.setContentCompressionResistancePriority( newValue ? .defaultHigh : .defaultLow, for: .horizontal)
+            title.setContentCompressionResistancePriority( newValue ? .defaultLow : .defaultHigh, for: .horizontal)
         }
         get {
-            return textField.placeholder
+            textField.contentCompressionResistancePriority(for: .horizontal) == .defaultHigh
         }
     }
 
+    private var customConstraints: [NSLayoutConstraint] = []
+
     var onTextFieldChange:((_:UITextField) -> Void)?	// set this from outside to get notified about textfield changes
 
+
+    public lazy var title: UILabel = {
+        let label = UILabel()
+        label.font = .preferredFont(forTextStyle: .body)
+        label.adjustsFontForContentSizeCategory = true
+        label.textColor = DcColors.defaultTextColor
+        label.lineBreakMode = .byTruncatingTail
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+
+    // use textFieldDelegate instead of textfield.delegate if you want to set a delegate from outside
+    public weak var textFieldDelegate: UITextFieldDelegate?
     lazy var textField: UITextField = {
         let textField = UITextField()
         textField.textAlignment = .right
-        // textField.enablesReturnKeyAutomatically = true
         textField.addTarget(self, action: #selector(textFieldChanged), for: .editingChanged)
+        textField.adjustsFontForContentSizeCategory = true
+        textField.font = .preferredFont(forTextStyle: .body)
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        textField.delegate = self
         return textField
     }()
 
+    public lazy var stackView: UIStackView = {
+        let view = UIStackView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.clipsToBounds = true
+        view.addArrangedSubview(title)
+        view.addArrangedSubview(textField)
+        view.axis = .horizontal
+        view.spacing = 10
+        return view
+    }()
+    
     init(description: String, placeholder: String, delegate: UITextFieldDelegate? = nil) {
-        super.init(style: .value1, reuseIdentifier: nil)
-        textLabel?.text = "\(description):"
+        placeholderVal = placeholder
+        super.init(style: .default, reuseIdentifier: nil)
+        title.text = "\(description):"
 
         // see: https://stackoverflow.com/a/35903650
         // this makes the textField respect the trailing margin of
         // the table view cell
         selectionStyle = .none
         setupViews()
-        textField.delegate = delegate
+        textFieldDelegate = delegate
         textField.placeholder = placeholder
+        preferValue = false
```

Comment:
in case there is not enough space for both title and text field content, the title is preferred by default. this was also the default before.